app/routes.py

```diff
@@ -1,2 +1,70 @@
-from flask import Blueprint
+from app import db
+from flask import Blueprint, jsonify, abort, make_response, request
+from .models.planet import Planet
 
+bp = Blueprint("planets", __name__, url_prefix="/planets")
+
+
+def validate_planet(planet_id):
+    try:
+        planet_id = int(planet_id)
+    except:
+        abort(make_response({"message": f"planet {planet_id} invalid"}, 400))
+
+    planet = Planet.query.get(planet_id)
+    if not planet:
+        abort(make_response({"message": f"planet {planet_id} not found"}, 404))
+
+    return planet
+
+def validate_planet_dict(request_body):
+    request_body = dict(request_body)
+    if not (request_body.get("name", False) and request_body.get("description", False)):
+        abort(make_response({'message': 'request body invalid; cannot create planet'}, 400))
+    if request_body.get("rings", 1) == 1:
+        abort(make_response({'message': 'request body invalid; cannot create planet'}, 400))
+
+@bp.route("", methods=["GET"])
+def handle_planets():
```

Comment:
It looks like you didn't get a chance to practice working with query parameters here (we were short on time). We could filter by a column, limit results, or order values, among many other possibilities!